<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <param name="/use_sim_time" value="true"/>

    <arg name="map_file" default="/home/mcp/catkin_ws/src/swarm/maps/maps_1.yaml"/>
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)">
        <param name="frame_id"  value="map"/>
    </node>

    <arg name="d1" default="drone1"/>
    <arg name="d2" default="drone2"/>
    <arg name="d3" default="drone3"/>
    <arg name="d4" default="drone4"/>


  <node pkg="nav_point_publisher" type="drone1_client.py" name="drone1_client_py" output="screen"/> 
  <node pkg="nav_point_publisher" type="drone2_client.py" name="drone2_client_py" output="screen"/> 
  <node pkg="nav_point_publisher" type="drone3_client.py" name="drone3_client_py" output="screen"/> 
  <node pkg="nav_point_publisher" type="drone4_client.py" name="drone4_client_py" output="screen"/> 


        <!--Call the navigation stack for drone1 -->
        <include file="$(find swarm)/launch/amcl.launch">
            <arg name="tf_prefix" value="drone1"/>
            <arg name="odom_frame_id" value="drone1/world"/> 
            <arg name="initial_pose_x" value="-40"/>
            <arg name="initial_pose_y" value="-42"/>
        </include>

     <!--Move Base for drone1 -->
     <group ns="drone1">
        <include file="$(find swarm)/launch/move_base_1.launch">
           <arg name="tf_prefix" value="drone1"/>
           <arg name="odom_frame_id" value="drone1/world"/>
        </include>
        </group>
        
        
        <!--Call the navigation stack for drone2 -->
        <include file="$(find swarm)/launch/amcl.launch">
            <arg name="tf_prefix" value="drone2"/>
            <arg name="odom_frame_id" value="drone2/world"/> 
            <arg name="initial_pose_x" value="-42"/>
            <arg name="initial_pose_y" value="40"/>
        </include>

     <!--Move Base for drone2 -->
     <group ns="drone2">
        <include file="$(find swarm)/launch/move_base_1.launch">
           <arg name="tf_prefix" value="drone2"/>
           <arg name="odom_frame_id" value="drone2/world"/>
        </include>
        </group>
        
        
        <!--Call the navigation stack for drone3 -->
        <include file="$(find swarm)/launch/amcl.launch">
            <arg name="tf_prefix" value="drone3"/>
            <arg name="odom_frame_id" value="drone3/world"/> 
            <arg name="initial_pose_x" value="120"/>
            <arg name="initial_pose_y" value="-44"/>
        </include>

     <!--Move Base for drone3 -->
     <group ns="drone3">
        <include file="$(find swarm)/launch/move_base_1.launch">
           <arg name="tf_prefix" value="drone3"/>
           <arg name="odom_frame_id" value="drone3/world"/>
        </include>
        </group>
      
        
        <!--Call the navigation stack for drone4 -->
        <include file="$(find swarm)/launch/amcl.launch">
            <arg name="tf_prefix" value="drone4"/>
            <arg name="odom_frame_id" value="drone4/world"/> 
            <arg name="initial_pose_x" value="120"/>
            <arg name="initial_pose_y" value="44"/>
        </include>

     <!--Move Base for drone4 -->
     <group ns="drone4">
        <include file="$(find swarm)/launch/move_base_1.launch">
           <arg name="tf_prefix" value="drone4"/>
           <arg name="odom_frame_id" value="drone4/world"/>
        </include>
        </group>

    
 
</launch>
